crates/ironrdp-cliprdr-format/src/bitmap.rs

```diff
@@ -166,11 +178,16 @@ impl BitmapInfoHeader {
     fn encode_with_size(&self, dst: &mut WriteCursor<'_>, size: u32) -> PduResult<()> {
         ensure_fixed_part_size!(in: dst);
 
+        check_invariant(self.width.abs() <= u16::MAX.into())
+            .ok_or_else(|| invalid_message_err!("biWidth", "width is too big"))?;
+        check_invariant(self.height.abs() <= u16::MAX.into())
+            .ok_or_else(|| invalid_message_err!("biHeight", "height is too big"))?;
```

Comment:
Invariants should be checked when constructing the struct (when decoding or calling new), so that remaining of the code may rely on this (encoding, casting…)